src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb

```diff
@@ -248,6 +248,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
         End Function
 
         Public Overrides Function TypeExpression(typeSymbol As ITypeSymbol) As SyntaxNode
+            If typeSymbol.IsSystemVoid Then
```

Comment:
Ah, great point about generating invalid code - that had completely slipped my mind.

OK, so I've updated the PR such that it only affects `SyntaxGenerator.MethodDeclaration(IMethodSymbol)`.

Thanks.